### DISCUSSION TITLE
Mockups Concept and Behavior of left panel Creating Symbols and using Symbols

### DISCUSSION THREAD

**Discussion by VikSin** [akiraux/Akira#474] [Ideas]
##  Symbols/Components

Symbols UI elements that allows you to reuse elements easily across your document’s Artboards and Pages, or even multiple documents. Sketch and Xd uses the term Symbol where as Figma use the term components.

### Creating and using symbols in Sketchapp
![sketch-color-override-3](https://user-images.githubusercontent.com/25245765/65311061-3403cc00-dbad-11e9-9344-9b836cb477ed.gif)

![1_z8Gzx9-iWouuoFwjm-2iqw](https://user-images.githubusercontent.com/25245765/65311092-467e0580-dbad-11e9-974f-355429b4784a.gif)

### Creating Symbols in Adobe Xd
![1_rV8CusMOg7ptrF37C6f-Lw](https://user-images.githubusercontent.com/25245765/65311688-81346d80-dbae-11e9-953a-9c696c17aa51.gif)

Both Sketch and Xd can create assets symbol, colors symbols, text, and all. 
Figma have separate Create Component and Create Style.

### Creating Component in Figma
![components_7](https://user-images.githubusercontent.com/25245765/65312438-213ec680-dbb0-11e9-8307-f57c7026f437.gif)
  
#### Creating styles in figma
![1_-CU2jmHvTVuTIBkEba5ZDQ](https://user-images.githubusercontent.com/25245765/65312900-0751b380-dbb1-11e9-84d8-b96e3965c9a5.gif)

## Left Panel When nothing is selected in canvas neither artboard nor objects

This optios will show
![Akiraux - 01 2 object mode](https://user-images.githubusercontent.com/25245765/65308540-eb95df80-dba7-11e9-9f4d-ba778ab51fbc.png)

### Mockups design Creating a symbol
![Akiraux - 01 2 object mode (1)](https://user-images.githubusercontent.com/25245765/65316372-729e8400-dbb7-11e9-97c9-bf12608ee26c.png)

### Using a symbol
![Akiraux - 01 2 object mode (2)](https://user-images.githubusercontent.com/25245765/65316411-8b0e9e80-dbb7-11e9-9fe5-a7b427db20ca.png)

Inserting Symbol from insert button it involved more clicks thats why separate library button to access for redusing click step  

### Another way of using and modfying symbol and their properties

![Akiraux - 01 2 object mode (3)](https://user-images.githubusercontent.com/25245765/65316522-c5783b80-dbb7-11e9-98cc-dfeab0c06b1e.png)
 
In Xd and Figma every time we have to switch from layer panel to asset panel and for accessing symbols . Sketch do better by having in menu.

**Comment by Alecaddd** [comment]
I love this! Thank you so much for kicking off the conversation.

**Name**
I personally prefer the term "Component" instead of "Symbol", as it makes more sens in terms of what it is and what it can encapsulate in it. A "Reusable Component" is also something commonly known in web development, which Akira is very much built for that.

**Create a Component**
The dedicated button in the Headerbar is :+1: 
Would you think it would be better to move it to the right of the "Insert button"? I think it should be part of the "Creation" group, so it might work better there.

I'm not sure about the dialog as I think the program should be smart enough to identify all the custom attributes getting stored in the component.
I can see the usefulness of allowing users to select which attributes should be part of a component, in order to create font styles alongside shape styles, so maybe we could sort of offer the dialog with a preselected list of options based on what the created component has.

**Components side panel**
I would put it on the right panel, below the layers panel, and make it collapsible for when users don't need it.
I like the concept anyway, and the various usability options.
How do you envision a symbol getting edited? Opening a separated view, an overlay artboard, or something else?

**Side note**
What is that beautiful Layout Grid Panel I see in the last mockup? That's really interesting?
Would you mind opening a dedicated issue to discuss that?
